components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLECPProviderServlet.java

```diff
@@ -0,0 +1,117 @@
+ * <li>The end point validates the SOAP bound ECP Request and extract the SAML Request from it.</li>
+ * <li>Then the servlet forwards the request to the https://<ip>:<port>/samlsso endpoint</li>
+ * </ol>
+ */
+
+public class SAMLECPProviderServlet extends HttpServlet {
+    private static Log log = LogFactory.getLog(SAMLECPProviderServlet.class);
+
+    protected void doGet(HttpServletRequest httpServletRequest,
+                         HttpServletResponse httpServletResponse) throws ServletException, IOException {
+        String soapFault = SAMLSOAPUtils.createSOAPFault("Unsupported Request POST"
+                , SAMLSOAPUtils.SOAP_FAULT_CODE_CLIENT);
+        log.error(soapFault);
+        PrintWriter out = httpServletResponse.getWriter();
+        httpServletResponse.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+        httpServletResponse.setContentType("text/html;charset=UTF-8");
+        out.print(soapFault);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        try {
+            handleRequest(req, resp);
+        } finally {
+            SAMLSSOUtil.removeSaaSApplicationThreaLocal();
+            SAMLSSOUtil.removeUserTenantDomainThreaLocal();
+            SAMLSSOUtil.removeTenantDomainFromThreadLocal();
+        }
+    }
+
+
+    private void handleRequest(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        try {
+            if (req.getHeader("Authorization") == null) {
+                String err = "Authorization Header cannot be Empty";
+                log.error(err);
+                throw new IdentitySAML2ECPException(err);
+            }
+        MessageFactory messageFactory = MessageFactory.newInstance();
+        InputStream inStream = req.getInputStream();
+        SOAPMessage soapMessage = messageFactory.createMessage(new MimeHeaders(), inStream);
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        soapMessage.writeTo(out);
+        String strMsg = new String(out.toByteArray(), Charset.forName("UTF-8"));
+
+        if (log.isDebugEnabled()) {
+            log.debug("ECP Request : " + strMsg);
+        }
+        String samlRequest = SAMLSOAPUtils.decodeSOAPMessage(soapMessage);
+        SamlSSORequestWrapper samlSSORequestWrapper = new SamlSSORequestWrapper(req);
+        samlSSORequestWrapper.setParameter(SAMLSSOConstants.SAML_REQUEST, samlRequest);
+        samlSSORequestWrapper.setParameter(SAMLSSOConstants.ECP_ENABLED, "true");
+        RequestDispatcher dispatcher = req.getRequestDispatcher("/samlsso");
+
+        log.info("Forwarding the ECP request to SAMLSSO Servlet");
+        dispatcher.forward(samlSSORequestWrapper, resp);
+
+        } catch (IdentitySAML2ECPException e) {
+            SAMLSOAPUtils.sendSOAPFault(resp, e.getMessage(), SAMLSOAPUtils.SOAP_FAULT_CODE_CLIENT);
```

Comment:
use a private method and call the send soap fault. 
exception throw and catch in same --- not

Reply:
> use a private method and call the send soap fault.
> exception throw and catch in same --- not
fixed.